src/rgw/rgw_op.cc

```diff
@@ -562,6 +566,67 @@ int rgw_build_object_policies(RGWRados *store, struct req_state *s,
   return ret;
 }
 
+void rgw_add_to_iam_environment(rgw::IAM::Environment& e, const std::string& key, const std::string& val){
+    e[key] = val;
```

Comment:
Not that this creates OR updates, so I think it's misleading WRT the same of the member function.